realm/src/androidTest/java/io/realm/RealmJsonTest.java

```diff
@@ -479,6 +493,37 @@ public void testCreateOrUpdateJsonObject() throws JSONException {
         assertEquals("bar", newObj.getColumnString());
     }
 
+    public void testCreateOrUpdateJsonObject_ignoreUnsetProperties() {
+        testRealm.beginTransaction();
+        AllTypesPrimaryKey obj = testRealm.createOrUpdateObjectFromJson(AllTypesPrimaryKey.class, "{ \"columnLong\":1, \"columnString\": \"foo\" }");
+        obj.setColumnBoolean(true);
```

Comment:
That line is just for setting up the test data. It might become more readable if I add the boolean to JSON. The interesting test is the one further below, where you call createOrUpdate again with the same JSON (without the boolean).